react-components/src/entry-client.tsx

```diff
@@ -1,17 +1,20 @@
-import React from 'react';
-import ReactDOM from 'react-dom/client';
-import App from './app';
-import './index.scss';
-import { BrowserRouter } from 'react-router-dom';
-import { Provider } from 'react-redux';
-import { store } from './store/store';
-
-ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
-  <React.StrictMode>
-    <BrowserRouter>
-      <Provider store={store}>
-        <App />
-      </Provider>
-    </BrowserRouter>
-  </React.StrictMode>
-);
+import React from 'react';
+import { hydrateRoot } from 'react-dom/client';
+import { BrowserRouter } from 'react-router-dom';
+import App from './app';
+import { Provider } from 'react-redux';
+import { store } from './store/store';
+import './index.scss';
+
+const root = document.getElementById('root') as HTMLElement;
+
+hydrateRoot(
+  root,
+  <React.StrictMode>
+    <Provider store={store}>

```

Comment:
you did not adopt Redux for SSR.

https://redux.js.org/usage/server-rendering#inject-initial-component-html-and-state
https://react-redux.js.org/api/provider#react-18-ssr-hydration